### ISSUE TITLE
Expose StakeProgram.Merge in solana/web3.js

### ISSUE THREAD

**Issue by CrieasofCarrots** [solana-labs-community/solana#163]
#### Problem
The `Merge` instruction from Stake Program is not exposed.

#### Summary of Changes
Adding the `Merge` instruction.

**Comment by jeffweshington** [comment]
Thanks for contributing!
Would you be willing to add test cases to https://github.com/solana-labs/solana/blob/v1.6/web3.js/test/stake-program.test.js ?

**Comment by CrieasofCarrots** [comment]
thanks for your review, I've updated this PR to make it a bit more complete.

Should I be able to run the E2E tests against `https://api.devnet.solana.com/`? It fails on my machine, is there any alternative for running those other than spinning up a Solana node?

**Comment by jeffweshington** [comment]
No, I don't think they're expected to pass against devnet at the moment (I believe the tests make some assumptions about ledger availability that may not be true of the devnet RPC nodes). Best approach is to run them against a local test validator: `npm run test:live-with-test-validator`
I do find the bpf loader test to be flaky for me locally.

Thanks for the update! This looks close. The only additional thing I'd like to see, if you're willing, is a case in the live staking actions
solana/web3.js/test/stake-program.test.ts
Line 326 in a9014ce
```rust
 it('live staking actions', async () => { 
```
ensuring that the merge transactions do succeed.
Probably easiest just to create 2 new undelegated stake accounts and merge those, or remerge the accounts that come out of split:
solana/web3.js/test/stake-program.test.ts
Line 473 in a9014ce
```rust
 // Split stake 
```

**Comment by t-neIson** [comment]
Also our commit linter is complaining about the commit message of eef74a9
Perhaps just squash it into a240d58 ?

<jeffweshington：>
* Squashed ✅
* Test case 😬
Happy to add a live test case, however my `npm run test:live-with-test-validator` is failing with the same issue described here #19006. I'll need to wait for the next `solana-test-validator` release as I don't want to build from source.

**Comment by sakeridge** [comment]
Thanks! No worries, I'll take over the live test case
Thank you very much for the contribution!
